os/board/common/crashdump.c

```diff
@@ -58,6 +78,185 @@
  * Private Functions
  ****************************************************************************/
 
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH) || defined(CONFIG_BOARD_COREDUMP_FLASH)
+static int get_dump_partname(char *part_name)
+{
+	FAR struct mtd_dev_s *mtd;
+	static char temp[MTD_PARTNAME_LEN + 1];
+	int index = 0;
+
+	do {
+		mtd = get_mtd_partition(index);
+		if (mtd) {
+			mtd_getpartitionname(mtd, temp);
+			if (!strncmp(temp, part_name, strlen(temp)))
+				return index;
+		}
+		index++;
+	} while (mtd);
+
+	return -1;
+}
+#endif
+
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+void coredump_to_flash(uint32_t cur_sp, void *tcb)
+{
+	uint32_t *ptr;
+	int32_t i;
+	uint32_t j;
+	uint32_t len = 0;
+	char *pbuf = (char *)coredump_buf;
+	int fd;
+	int index = 0;
+	int nbytes = 0;
+	char coredump_partname[MTD_PARTNAME_LEN + 1];
+
+	index = get_dump_partname("coredump");
+	if (index < 0) {
+		lldbg("partition index is wrong for coredump part name\n");
+		return errno;
+	}
+
+	snprintf(coredump_partname, MTD_PARTNAME_LEN + 1, "/dev/mtdblock%d", index);
+	lldbg("Coredump to partition = %s\n", coredump_partname);
+
+	sched_lock();
+	fd = open(coredump_partname, O_WRONLY);
```

Comment:
This execution is in assert state. We never guarantee that file operation is working well at that time.
Let's use low level API directly. That's why ramdump via uart uses up_xxx APIs.

Reply:
Hi, we are using mtd driver now for partition read and write, if we use low level API it might be chip specific.

Reply:
added code to access low level API